### PR TITLE
MARKET-1396 Cropper component

### PR DESCRIPTION
**Pull request by serhmel**
thumbnail
Screenshot_4 
Screenshot_7

UI Builder Preview:
cropper_ui_preview

### CODE REVIEW

**Review by Saharij — COMMENTED** [review]
LGTM🔥

File: components/bl-cropper-component/src/cropper-components.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+  const onDragOver = e => {
+    e.stopPropagation();
+    e.preventDefault();
+  };
+
+  const onDragEnter = e => {
+    e.preventDefault();
+    e.stopPropagation();
+  };
```
Comment (v-excelsior) [inline components/bl-cropper-component/src/cropper-components.js:14]: Could you explain how it works?

File: components/bl-cropper-component/src/helpers.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import { useEffect, useRef } from 'react';
+
+import Cropper from './lib/cropper.min';
+
+export function UseCropperLibrary(component, eventHandlers, imageRef, image) {
```
Comment (v-excelsior) [inline components/bl-cropper-component/src/helpers.js:5]: why `Use` not `use`?

File: components/bl-cropper-component/src/helpers.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      toggleDragModeOnDblclick,
+      minContainerWidth,
+      minContainerHeight,
+      minCanvasWidth,
+      minCanvasHeight,
+      minCropBoxWidth,
+      minCropBoxHeight,
+      ready() {
+        onReady();
+      },
```
Comment (v-excelsior) [inline components/bl-cropper-component/src/helpers.js:47]: `ready: () => onReady()` ?

File: components/bl-cropper-component/src/cropper-components.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    const newScaleY = -cropperRef.current.getImageData().scaleY;
+
+    cropperRef.current.scaleY(newScaleY || -1);
+  };
+
+  const toolbar = [
+    [{ label: 'zoom_in', onClick: zoomIn }, { label: 'zoom_out', onClick: zoomOut }],
+    [{ label: 'rotate_left', onClick: rotateLeft }, { label: 'rotate_right', onClick: rotateRight }],
+    [{ label: 'swap_horiz', onClick: scaleX }, { label: 'swap_vert', onClick: scaleY }],
+  ];
```
Comment (v-excelsior) [inline components/bl-cropper-component/src/cropper-components.js:86]: useMemo?

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-cropper-component/src/cropper-components.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+      <div className="buttons-wrapper">
+        <Button onClick={ saveImage } label="Save" visibility={ saveButtonVisibility }/>
+        <Button onClick={ downloadImage } label="Download" visibility={ downloadButtonVisibility }/>
+      </div>
+    </div>
+  );
+}
+
+export function FooterToolbar({ cropperRef, image, toolbarVisibility }) {
+  const toolbar = useMemo(() => {
```
Comment (v-excelsior) [inline components/bl-cropper-component/src/cropper-components.js:69]: :thinking:

File: components/bl-cropper-component/src/cropper-components.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+function ButtonsGroup({ buttons }) {
+  return (
+    <div className="buttons-group">
+      { buttons.map((button, index) => (
+        <Button
+          className="material-icons-round"
+          onClick={ button.onClick }
+          label={ button.label }
+          visibility={ true }
```
Comment (v-excelsior) [inline components/bl-cropper-component/src/cropper-components.js:113]: `visibility={ true }` is it correct?

File: components/bl-cropper-component/styles/index.less [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+    }
+  }
+
+  .footer-toolbar {
+    display: flex;
+    justify-content: center;
+    flex-wrap: wrap;
+    align-items: center;
+
+    .buttons-group {
```
Comment (v-excelsior) [inline components/bl-cropper-component/styles/index.less:106]: I will offer this variant of button group styling
image

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-cropper-component/src/cropper-components.js (hunk 1 above)
Comment (v-excelsior) [reply components/bl-cropper-component/src/cropper-components.js:16]: Why they are called in different order?

```
e.stopPropagation();
e.preventDefault();
...
e.preventDefault();
e.stopPropagation();
```

File: components/bl-cropper-component/src/cropper-components.js (hunk 6 above)
Comment (serhmel) [reply components/bl-cropper-component/src/cropper-components.js:113]: Yes, that is correct. 
The `visibility `attribute on the `Button` component is needed in `HeaderToolbar`, where I also use this component. In `ButtonsGroup` it should be `true`.

File: components/bl-cropper-component/src/cropper-components.js (hunk 1 above)
Comment (serhmel) [reply components/bl-cropper-component/src/cropper-components.js:16]: I think the order doesn't matter here.

File: components/bl-cropper-component/styles/index.less (hunk 7 above)
Comment (serhmel) [reply components/bl-cropper-component/styles/index.less:106]: Screenshot_12

**Pull request merged**